MetaMask/metamask-mobile
# feat: @lavamoat/git-safe-dependencies

## **Description**

Quality Quest 2024 ft. @naugtur and @tommasini b2b

TODO
- [x] add package
- [x] integrate in CI
- [x] update depcheck
- [x] socket

## **Related issues**

- Fixes: https://github.com/MetaMask/mobile-planning/issues/2079

## **Manual testing steps**

1. Locally update package.json to `"react-native-tcp": "aprock/react-native-tcp#11/head"`
2. Run `yarn git-safe-dependencies`
3. There should be an error message
4. Revert package.json to the correct `"react-native-tcp": "aprock/react-native-tcp#98fbc801f0586297f16730b2f4c75eef15dfabcd",`
5. Run `yarn git-safe-dependencies`
6. great
7. success

## **Screenshots/Recordings**

### **Before**

e.g. package.json > `"react-native-tcp": "aprock/react-native-tcp#11/head",`

![image](https://github.com/user-attachments/assets/f054cda3-ce19-4b64-a84d-e1f92397d2be)

### **After**

fixed to: package.json > `"react-native-tcp": "aprock/react-native-tcp#98fbc801f0586297f16730b2f4c75eef15dfabcd",`
in https://github.com/MetaMask/metamask-mobile/pull/12595

![image](https://github.com/user-attachments/assets/a3340650-743f-48f8-a4bc-b1dcfd81d243)

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
